core/src/main/scala/org/apache/spark/storage/ExternalBlockStore.scala

```diff
@@ -113,7 +143,13 @@ private[spark] class ExternalBlockStore(blockManager: BlockManager, executorId:
   }
 
   override def getValues(blockId: BlockId): Option[Iterator[Any]] = {
-    getBytes(blockId).map(buffer => blockManager.dataDeserialize(blockId, buffer))
+    try {
+      externalBlockManager.flatMap(_.getValues(blockId))
+    } catch {
+      case NonFatal(t) =>
+        logError(s"error in getValues from $blockId", t)
```

Comment:
This is really opinion too, but if it's not fatal, I think it could be a warning instead of error.